### PR TITLE
gh-116984: Install mimalloc headers

### PR DESCRIPTION
**Pull request by oraluben**
This PR installs mimalloc headers and allow extensions that access `pycore_*.h` continue working.

Such extensions need to explicitly add `internal/mimalloc` to its header search path.

* Issue: gh-116984

### CODE REVIEW

**Review by colesbury — COMMENTED** [review]
I think we should only install the headers if Python is built with mimalloc (i.e., not configured with `--without-mimalloc`).

File: configure.ac [hunk 1]
```diff
@@ -4752,7 +4752,7 @@ elif test "$disable_gil" = "yes"; then
 fi
 
 AC_MSG_RESULT([$with_mimalloc])
-AC_SUBST([WITH_MIMALLOC])
+AC_SUBST([INSTALL_MIMALLOC], $with_mimalloc)
```
Comment (erlend-aasland) [inline configure.ac]:
```suggestion
AC_SUBST([INSTALL_MIMALLOC], [$with_mimalloc])
```

**Review by colesbury — APPROVED** [review]
This looks good to me, but I'll wait for other people to leave their feedback.

As written in the [devguide](https://devguide.python.org/developer-workflow/c-api/#the-internal-api), for users, the internal API is sometimes "the best workaround for a thorny problem", so we want it to compile successfully in an Python installation.

At the same time, we don't always prefix names with `Py` or `_Py` in the internal API, so including CPython internal headers can lead to language level name conflicts. In this case, if an extensions includes our internal headers *and* bundles mimalloc, they could run into problems. While not ideal, I don't think that's a reason not to make the internal headers compile at all in installations.

**Comment by oraluben** [comment]
Hi @vstinner , would you like to help landing this?

**Comment by erlend-aasland** [comment]
Thanks @oraluben and @colesbury!

**Pull request merged**